Add constructors for feedback logs

carolynvs opened the pull request:
The feedback system was originally designed for reporting feedback as GOPATH was scanned, which didn’t work well for logging from the importers. This moves more of the logic around what should be logged back onto the caller and simplifies the `ConstraintFeedback.LogFeedback` function to print what it is given.

This is broken off from #715.

sdboyer commented on internal/feedback/feedback.go:
```diff
@@ -25,23 +28,50 @@ const DepTypeImported = "imported dep"
+		cf.Revision = vt.Underlying().String()
+	case gps.Revision:
+		cf.Revision = vt.String()
+	}
```
It should be, as a practical matter, impossible to have a `gps.UnpairedVersion` here. But it's still logically possible, so we need to have a case for it all the same. 

```go
case gps.UnpairedVersion:
	cf.LockedVersion = vt.String()
```

carolynvs commented:
* Rebased and squashed
* Updated the new godep importer to use the feedback constructor
* Tweaked existing feedback tests to exercise the constructors

sdboyer commented:
LGTM! 🎉
